install/tools/istio_vm_common.sh

```diff
@@ -0,0 +1,216 @@
+#!/usr/bin/env bash
+
+# Helper functions for extending the cluster with external VMs.
+
+# Script can be sourced in other files or used from tools like ansible.
+# Environment variables used:
+
+# ISTIO_FILES - directory where istio artifacts are downloaded, default to current dir
+
+# Initialize internal load balancers to access K8S DNS and Istio Pilot, Mixer, CA.
+# Must be run once per cluster.
+function istioInitILB() {
+   local NS=${ISTIO_NAMESPACE:-istio-system}
+
+cat <<EOF | kubectl apply -n $NS -f -
+apiVersion: v1
+kind: Service
+metadata:
+  name: istio-pilot-ilb
+  annotations:
+    cloud.google.com/load-balancer-type: "internal"
+  labels:
+    istio: pilot
+spec:
+  type: LoadBalancer
+  ports:
+  - port: 8080
+    protocol: TCP
+  selector:
+    istio: pilot
+EOF
+cat <<EOF | kubectl apply -n kube-system -f -
+apiVersion: v1
+kind: Service
+metadata:
+  name: dns-ilb
+  annotations:
+    cloud.google.com/load-balancer-type: "internal"
+  labels:
+    k8s-app: kube-dns
+spec:
+  type: LoadBalancer
+  ports:
+  - port: 53
+    protocol: UDP
+  selector:
+    k8s-app: kube-dns
+EOF
+
+cat <<EOF | kubectl apply -n $NS -f -
+apiVersion: v1
+kind: Service
+metadata:
+  name: mixer-ilb
+  annotations:
+    cloud.google.com/load-balancer-type: "internal"
+  labels:
+    istio: mixer
+spec:
+  type: LoadBalancer
+  ports:
+  - port: 9091
+    protocol: TCP
+  selector:
+    istio: mixer
+EOF
+
+cat <<EOF | kubectl apply -n $NS -f -
+apiVersion: v1
+kind: Service
+metadata:
+  name: istio-ca-ilb
+  annotations:
+    cloud.google.com/load-balancer-type: "internal"
```

Comment:
could the annotation/name be kept generic, e.g. not platform specific?

Reply:
Unfortunately that's the magic syntax that is required. We can add multiple annotations ( to cover the other platforms that need magic strings ).

Reply:
See above - hopefully a 'standard' k8s emerges, but until then we'll need one for each platform.